src/javascript/app/pages/trade/purchase.js

```diff
@@ -63,17 +65,37 @@ const Purchase = (() => {
             container.style.display = 'block';
             message_container.hide();
             confirmation_error.show();
+
+            const required_api_calls = ['authorize', 'website_status', 'landing_company'];
+            const is_logged_in = Client.isLoggedIn();
+            if (is_logged_in) {
+                required_api_calls.push('get_account_status');
+            }
+
             let message = error.message;
-            if (/RestrictedCountry/.test(error.code)) {
-                let additional_message = '';
-                if (/FinancialBinaries/.test(error.code)) {
-                    additional_message = localize('Try our [_1]Volatility Indices[_2].', [`<a href="${urlFor('get-started/binary-options', 'anchor=volatility-indices#range-of-markets')}" >`, '</a>']);
-                } else if (/Random/.test(error.code)) {
-                    additional_message = localize('Try our other markets.');
+            BinarySocket.wait(...required_api_calls).then(() => {
+                const is_eu = isEuCountry();
+                const is_maltainvest = Client.get('landing_company_shortcode') === 'maltainvest';
+                const is_no_mfp_client_error = /NoMFProfessionalClient/.test(error.code);
+               
+                if (is_eu && is_no_mfp_client_error) {
+                    if (is_logged_in && is_maltainvest) {
+                        const is_professional = (State.getResponse('get_account_status.status') || []).includes('professional');
+                        if (!is_professional) {
```

Comment:
same logic here, if the API is already telling us that client can't purchase this client because of `NoMFProfessionalClient`, i.e. current client is a non-professional MF, we don't need to check it again in FE if current client is professional of not, so this if condition can be removed entirely as well.